extra_data/exceptions.py

```diff
@@ -6,11 +6,13 @@ class FileStructureError(Exception):
 
 
 class SourceNameError(KeyError):
-    def __init__(self, source):
+    # If `custom_message=True`, then `source` is treated as the full error message
+    def __init__(self, source, custom_message=False):
         self.source = source
+        self.custom_message = custom_message
 
     def __str__(self):
-        return (
+        return self.source if self.custom_message else (
```

Comment:
Shouldn't this be `return self.custom_message if self.custom_message else ...` 🤔

Reply:
Nope, `custom_message` was originally just a boolean flag so in this case `self.source` would be the custom message. Possibly better named as `use_custom_message` :sweat_smile:

Reply:
Oh... so `self.source` will be any text and `custom_message` is just a flag to display it instead?

I wonder if I would rename it then to avoid confusion. I doubt it was ever called with argument names, so it should still continue to work the old way. Sorry, the above comment concerning calling by keyword is then meant in a different way.